testing/citest/tests/openstack_smoke_test.py

```diff
@@ -0,0 +1,265 @@
+             'rules': jp.STR_SUBSTR("protocol='udp'")
+                      and jp.STR_SUBSTR("port_range_min='10'")
+                      and jp.STR_SUBSTR("port_range_max='65530'")}),
+         jp.DICT_MATCHES({
+             'rules': jp.STR_SUBSTR("protocol='icmp'")
+                      and jp.STR_SUBSTR("port_range_min='12'")
+                      and jp.STR_SUBSTR("port_range_max='9'")})]))
+
+    payload = self.agent.make_json_payload_from_kwargs(
+        job=job, description=' Test - create security group for {app}'.format(
+            app=self.TEST_APP),
+        application=self.TEST_APP)
+
+    return st.OperationContract(
+        self.new_post_operation(
+            title='create_security_group', data=payload,
+            path='applications/{app}/tasks'.format(app=self.TEST_APP)),
+        contract=builder.build())
+
+  def delete_a_security_group(self):
+    """Creates OsContract for deleteServerGroup.
+
+    To verify the operation, we just check that the spinnaker security group
+    for the given application was deleted.
+    """
+    #Get ID of the created security group
+    os_agent = os.OsAgent(None)
+    data = os_agent.get_resource('security group', self.TEST_SECURITY_GROUP)
+    security_group_id = data['id']
+
+    payload = self.agent.make_json_payload_from_kwargs(
+        job=[{
+            "Provider": "openstack",
+            "id": security_group_id,
+            "region": self.bindings['OS_REGION_NAME'],
+            "regions": [self.bindings['OS_REGION_NAME']],
+            "account": self.bindings['SPINNAKER_OS_ACCOUNT'],
+            "securityGroupName": self.TEST_SECURITY_GROUP,
+            "cloudProvider": "openstack",
+            "type": "deleteSecurityGroup",
+            "user": self.bindings['TEST_OS_USERNAME']
+        }],
+        application=self.TEST_APP,
+        description='Delete Security Group: : ' + self.TEST_SECURITY_GROUP)
+
+    builder = os.OsContractBuilder(self.os_observer)
+    (builder.new_clause_builder(
+        'Security Group Deleted', retryable_for_secs=30)
+     .show_resource('security group', self.TEST_SECURITY_GROUP,
+                    no_resources_ok=True)
+     .excludes_path_eq('name', self.TEST_SECURITY_GROUP)
+    )
+    return st.OperationContract(
+        self.new_post_operation(
+            title='delete_security_group', data=payload,
+            path='applications/{app}/tasks'.format(app=self.TEST_APP)),
+        contract=builder.build())
+
+class OpenStackSmokeTest(st.AgentTestCase):
+  """The test fixture for the OpenStackSmokeTest.
+
+  This is implemented using citest OperationContract instances that are
+  created by the OpenStackSmokeTestScenario.
+  """
+  # pylint: disable=missing-docstring
+  @property
+  def scenario(self):
+    return citest.base.TestRunner.global_runner().get_shared_data(
+        OpenStackSmokeTestScenario)
+
+  def test_a_create_app(self):
+    self.run_test_case(self.scenario.create_app())
+
+  def test_z_delete_app(self):
```

Comment:
list the tests in the order intended to be run
(and keep the names lexographic to that order as well)